### ISSUE TITLE
Update build and install instructions

### ISSUE THREAD

**Issue by richlander** [dotnet/coreclr#2443]
There are multiples sets of instructions. They don't seem like the ones we should have as we approach RTM.

Here are the experiences that I think we should make easy:

1. Build the world! Build the runtime (CoreCLR or CoreRT), CoreFX, the language compiler (Roslyn or F#) and CLI from source. If using CoreRT, also build the native compiler (CoreRT or LLILC) from source.
2. Build just the runtime. Rely on NuGet for CoreFX. Using CLI should be optional.
3. Build CoreFX (in part or whole.) Rely on NuGet for the runtime. Using CLI should be optional.

You should have a working dev environment. That's something that has been missing. You should be able to compile apps and run unit tests. You should be able to use VS Code as an editor.  There should be no Mono dependency or packages.config.

We now have CLI as a good driver. That's a major step forward. I am guessing that there are folks that will want an experience more low-level than CLI. We should determine if this is the case to ensure that there are instructions for their use case.

What else do we need good instructions for?

There is also a q of which documents are advertised in the root README.md:

- The DNX docs should all be removed from the Documentation directory. 
- The DNX references in the front-page README.me should be replaced with new documents.
- The "Build from source" docs should be the ones mentioned above. The "Get Binaries" links should reference docs in the CLI repo.

#2386 satisfies some of this. I wanted to raise this issue in a wholistic way so that we can define what we all want. Just jumping over to CLI as an extension of the compiler doesn't quite seem right.

/cc @ellismg, @blackdwarf, @kangaroo

**Comment by OtherCrashOverride** [comment]
> What else do we need good instructions for?

Remote deploy and debug from Visual Studio on Windows to a instance of CoreCLR running on Mac/Linux.

> We now have CLI as a good driver. That's a major step forward. I am guessing that there are folks that will want an experience more low-level than CLI.

Should also explain what that means in English. Is CLI = Command Line Interface, Common Language Infrastructure, somethig else entirely? If its Command Line Interface, what is an example of something that is lower-level? Punch cards come to mind. :wink:

**Comment by blackdwarf** [comment]
@richlander agreed on all above. Documentation on how to build everything from source sounds good as well. What did you mean by more low-level than the CLI?

@OtherCrashOverride CLI == Command Line Interface in this specific instance.

**Comment by richlander** [comment]
I mean using the c# compiler and corerun directly.

**Comment by OtherCrashOverride** [comment]
> CLI == Command Line Interface in this specific instance. 

Actually, I think it means:

CLI == ".NET Command Line Interface" == DNX == https://github.com/dotnet/cli

CLI != "Dos Prompt"
CLI != "Command Prompt"
CLI != "Terminal"

**Comment by OtherCrashOverride** [comment]
As there seems to be some confusion, maybe there should be an introductory document that clearly explains what all these terms mean in the context of the documentation.  Additionally, terms should be used consistently across all documentation.  For example, is DNX even a thing anymore? The original post refers to both CLI and DNX.

Its like how everything is called Windows 10, but you can not run Visual Studio on Windows 10. It only works on Windows 10, not Windows 10.

(Translation of above sentence:  Its like how everything is called Windows 10, but you can not run Visual Studio on Windows 10 mobile. It only works on Windows 10 desktop, not Windows 10 IoT.)

**Comment by richlander** [comment]
Sorry for the lack of clarity @OtherCrashOverride. Here's a few documents that we published that should help: https://github.com/dotnet/core/blob/master/roadmap.md#technology-roadmaps

Main takeaway is that DNX != CLI.

**Comment by blackdwarf** [comment]
@OtherCrashOverride sorry for the confusion, I was thinking primarily to your point whether CLI refers to Common Language Infrastructure. :) 

@richlander so, the lower-level approach...that would require getting the dependencies, getting the ref assemblies and compiling using csc/vbc and then extracting the implementation assemblies and placing them near the DLL and corerun, correct?

**Comment by richlander** [comment]
@blackdwarf yes! There is also the q of how that works if you build from source. The top-level q is whether folks want that lower level experience documented and supported.

**Comment by OtherCrashOverride** [comment]
Has there been any progress on this?  I am particularly interested in:
> Remote deploy and debug from Visual Studio on Windows to a instance of CoreCLR running on Mac/Linux.
